[WIP] feat: rewrite of ngx-sub-form without inheritance

maxime1992 opened the pull request:
See commit description for more.

tyroneneill commented on projects/ngx-sub-form/src/lib/new/helpers.ts:
```diff
@@ -0,0 +1,187 @@
+
+export const patchClassInstance = (componentInstance: any, obj: Object) => {
+  Object.entries(obj).forEach(([key, newMethod]) => {
+    componentInstance[key] = newMethod;
```
I think it would be safer to check if the instance has any of the methods implemented and decide to error or incorporate both calls.

tyroneneill replied:
What happens if the instance already has one of the hooks defined? is it on purpose we do not use `safelyPatchClassInstance` ?

maxime1992 commented:
I'm going to close #176 in favor of #188 which is the same of except that I renamed the branch to go around the publishing error with the not supported format for the branch feat/rewrite which I renamed in the new branch to feat-rewrite.

@tyroneneill I'll keep your comments in mind before merging and have a look at some point :+1:
